File: niuniu-take-out-springboot3/server/src/main/java/common/niuniu/service/DishService.java
package common.niuniu.service;

import common.niuniu.dto.DishDTO;
import common.niuniu.dto.DishPageDTO;
import common.niuniu.entity.Dish;
import common.niuniu.result.PageResult;
import common.niuniu.vo.DishVO;

import java.util.List;

public interface DishService {
    void addDishWithFlavor(DishDTO dishDTO);

    PageResult getPageList(DishPageDTO dishPageDTO);

    DishVO getDishWithFlavorById(Integer id);

    void updateDishWithFlavor(DishDTO dishDTO);

    void deleteBatch(List<Integer> ids);

    void onOff(Integer id);

    List<DishVO> getDishesWithFlavorById(Dish dish);
}

File: niuniu-take-out-kotlin/server/src/main/kotlin/common/niuniu/service/DishService.kt
package common.niuniu.service

import common.niuniu.dto.DishDTO
import common.niuniu.dto.DishPageDTO
import common.niuniu.po.Dish
import common.niuniu.result.PageResult
import common.niuniu.vo.DishVO

interface DishService {
    fun addDishWithFlavor(dishDTO: DishDTO?)

    fun getPageList(dishPageDTO: DishPageDTO?): PageResult?

    fun getDishWithFlavorById(id: Int?): DishVO?

    fun updateDishWithFlavor(dishDTO: DishDTO?)

    fun deleteBatch(ids: List<Int?>?)

    fun onOff(id: Int?)

    fun getDishesWithFlavorById(dish: Dish?): List<DishVO>
}
